compiler/rustc_hir_analysis/src/hir_ty_lowering/bounds.rs

```diff
@@ -698,6 +864,27 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
     }
 }
 
+fn push_lang_item_trait<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    bounds: &mut Vec<(ty::Clause<'tcx>, Span)>,
+    ty: Ty<'tcx>,
+    lang_item: LangItem,
+    span: Span,
+) {
+    assert_eq!(lang_item.target(), rustc_hir::Target::Trait);
+    if lang_item == LangItem::Sized {
+        let sized_def_id = tcx.require_lang_item(LangItem::Sized, Some(span));
+        let trait_ref = ty::TraitRef::new(tcx, sized_def_id, [ty]);
+        // Preferable to put this obligation first, since we report better errors for sized ambiguity.
+        bounds.insert(0, (trait_ref.upcast(tcx), span));
```

Comment:
can you instead only do the `if lang_:item == LangItem::Sized` for the final push/insert instead of duplicating the trait_ref creation?

Reply:
It may be worth deleting this branch completely and leaving only "insert", since this only affects the diagnostic of experimental traits.